.golangci.yml

```diff
@@ -1,17 +1,37 @@
 linters:
   enable-all: true
   disable:
+    - contextcheck
```

Comment:
@stmcginnis @srm09 WDYT could we maybe change this to specify the list of enabled linters instead?
I think it would make it way easier to see which linters are enabled (and also potentially to compare it with other providers to pick up new linters)

Reply:
We could, but my fear would be we would miss some useful linters that we just aren't aware of yet. Enabling all of them and then disabling explicitly helps, in my opinion, by making it explicit what we are deciding we do not want to lint on.

If/when we upgrade the version of golangci-lint that we're using, it has the risk that it pulls in more linters that we were not enforcing before. But that can be useful to realize there is maybe something we should consider that would improve our code quality. Personally, I would rather go through that and make the explicit decision that we are not going to enforce something, than to risk not knowing about issues because we're ignoring everything by default.

Happy to change this though if the general consensus is to do it the other way.

Reply:
Yup agree, your perspective sounds very reasonable. It's essentially a trade-off between "seeing" what's enabled vs missing out on new linters with newer golangci-lint versions. I'm fine with both, just thought I'd ask :)

Reply:
@vincepri We definitely don't have to do it the same way in all provider / core CAPI, but I wonder what's your take on it?

Reply:
I've opened a Slack thread: https://kubernetes.slack.com/archives/C8TSNPY4T/p1642768320122200

Reply:
I had a discussion with @srm09, for CAPV it would be good idea to align with CAPI. So we would have to change it to enable list in that case.